dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/codec/ExchangeCodec.java

```diff
@@ -378,7 +379,11 @@ protected void encodeData(ObjectOutput out, Object data) throws IOException {
     }
 
     private void encodeEventData(ObjectOutput out, Object data) throws IOException {
-        out.writeObject(data);
+        if(data != null) {
+            out.writeObject(data);

```

Comment:
would it be better to implement it like this?

```java
    @Deprecated
    protected Object decodeHeartbeatData(Channel channel, ObjectInput in) throws IOException {
        return Response.HEARTBEAT_EVENT;
    }
```

Reply:
Yes, it's better to implement.

Reply:
encode I suggest or judge. To prevent accidents
2.7.3 vesion I see ProtostuffObjectOutput.writeObject ，
In the method, I see “obj == null”

Reply:
I think it's fine, but I suggest we still need to make encode heartbeat to no-op, what do you think:

```java
    @Deprecated
    protected void encodeHeartbeatData(ObjectOutput out, Object data) throws IOException {
        // no op since heartbeat_event is null
    }
```